test/extended/builds/valuefrom.go

```diff
@@ -0,0 +1,141 @@
+		failedDockerBuildValueFrom     = filepath.Join(valueFromBaseDir, "failed-docker-build-value-from-config.yaml")
+		oc                             = exutil.NewCLI("build-valuefrom", exutil.KubeConfigPath())
+	)
+
+	g.JustBeforeEach(func() {
+		g.By("waiting for builder service account")
+		err := exutil.WaitForBuilderAccount(oc.KubeClient().Core().ServiceAccounts(oc.Namespace()))
+		o.Expect(err).NotTo(o.HaveOccurred())
+		g.By("waiting for openshift namespace imagestreams")
+		err = exutil.WaitForOpenShiftNamespaceImageStreams(oc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+	})
+
+	g.It("should successfully process valueFrom in s2i build strategy environment variables", func() {
+
+		g.By("creating test image stream")
+		err := oc.Run("create").Args("-f", stiImageStreamFixture).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("creating test secret")
+		err = oc.Run("create").Args("-f", secretFixture).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("creating test configmap")
+		err = oc.Run("create").Args("-f", configmapFixture).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("creating test successful build config")
+		err = oc.Run("create").Args("-f", successfulSTIBuildValueFrom).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("starting test build")
+		br, err := exutil.StartBuildAndWait(oc, "mys2itest")
+		o.Expect(err).NotTo(o.HaveOccurred())
+		br.AssertSuccess()
+
+		o.Expect(br.Logs()).To(o.ContainSubstring("FIELDREF_ENV=mys2itest-1"))
+		o.Expect(br.Logs()).To(o.ContainSubstring("CONFIGMAPKEYREF_ENV=myvalue"))
+		o.Expect(br.Logs()).To(o.ContainSubstring("SECRETKEYREF_ENV=developer"))
+
+	})
+
+	g.It("should successfully process valueFrom in docker build strategy environment variables", func() {
+
+		g.By("creating test image stream")
+		err := oc.Run("create").Args("-f", dockerImageStreamFixture).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("creating test secret")
+		err = oc.Run("create").Args("-f", secretFixture).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("creating test configmap")
+		err = oc.Run("create").Args("-f", configmapFixture).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("creating test successful build config")
+		err = oc.Run("create").Args("-f", successfulDockerBuildValueFrom).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("starting test build")
+		br, err := exutil.StartBuildAndWait(oc, "mydockertest")
+		o.Expect(err).NotTo(o.HaveOccurred())
+		br.AssertSuccess()
+
+		o.Expect(br.Logs()).To(o.ContainSubstring("\"FIELDREF_ENV\" \"mydockertest-1\""))
+		o.Expect(br.Logs()).To(o.ContainSubstring("\"CONFIGMAPKEYREF_ENV\" \"myvalue\""))
+		o.Expect(br.Logs()).To(o.ContainSubstring("\"SECRETKEYREF_ENV\" \"developer\""))
+
+	})
+
+	g.It("should fail processing ResourceFieldRef and value/valueFrom in s2i build strategy environment variables", func() {
```

Comment:
this should be a validation unit test (which you already have), no need for a full extended test for it.